src/main/java/services/RestoreFactory.java

```diff
@@ -0,0 +1,88 @@
+package services;
+
+import auth.HqAuth;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Created by willpride on 9/21/16.
+ */
+public class RestoreFactory {
+    @Value("${commcarehq.host}")
+    private String host;
+
+    private String asUsername;
+    private String username;
+    private String domain;
+    private HqAuth hqAuth;
+
+    private final Log log = LogFactory.getLog(RestoreFactory.class);
+
+
+    public String getRestoreXml() {
+        String restoreUrl;
+        if (asUsername == null) {
+            restoreUrl = getRestoreUrl(host, domain);
+        } else {
+            restoreUrl = getRestoreUrl(host, domain, asUsername);
+        }
+
+        log.info("Restoring from URL " + restoreUrl);
+        return getRestoreXmlHelper(restoreUrl, hqAuth);
+    }
+
+    private static String getRestoreXmlHelper(String restoreUrl, HqAuth auth) {
+        RestTemplate restTemplate = new RestTemplate();
+        ResponseEntity<String> response =
+                restTemplate.exchange(restoreUrl,
+                        HttpMethod.GET,
+                        new HttpEntity<String>(auth.getAuthHeaders()), String.class);
+        return response.getBody();
+    }
+
+    public static String getRestoreUrl(String host, String domain){
+        return host + "/a/" + domain + "/phone/restore/?version=2.0";
+    }
+
+    public String getRestoreUrl(String host, String domain, String username) {
+        return host + "/hq/admin/phone/restore/?as=" + username + "@" +
+                domain + ".commcarehq.org&version=2.0&raw=true";
```

Comment:
minor thing that we probably don't have to worry about right now but this'll fail if you try to restore as another web user. this could happen if you're a web user and want to edit a form that was submitted by a web user. probably pretty rare